src/cascadia/TerminalControl/XamlUiaTextRange.cpp

```diff
@@ -0,0 +1,176 @@
+    {
+        auto self = winrt::get_self<XamlUiaTextRange>(pTargetRange);
+
+        int32_t returnVal;
+        THROW_IF_FAILED(_uiaProvider->Compare(self->_uiaProvider.get(), &returnVal));
+        return returnVal;
+    }
+
+    void XamlUiaTextRange::ExpandToEnclosingUnit(Windows::UI::Xaml::Automation::Text::TextUnit unit) const
+    {
+        THROW_IF_FAILED(_uiaProvider->ExpandToEnclosingUnit(static_cast<::TextUnit>(unit)));
+    }
+
+    Windows::UI::Xaml::Automation::Provider::ITextRangeProvider XamlUiaTextRange::FindAttribute(int32_t textAttributeId,
+                                                                                                winrt::Windows::Foundation::IInspectable val,
+                                                                                                bool searchBackward)
+    {
+        // we don't support this currently
+        return nullptr;
+    }
+
+    Windows::UI::Xaml::Automation::Provider::ITextRangeProvider XamlUiaTextRange::FindText(winrt::hstring text,
+                                                                                           bool searchBackward,
+                                                                                           bool ignoreCase)
+    {
+        // TODO GitHub #605: Search functionality
+        // we need to wrap this around the UiaTextRange FindText() function
+        // but right now it returns E_NOTIMPL, so let's just return nullptr for now.
+        return nullptr;
+    }
+
+    winrt::Windows::Foundation::IInspectable XamlUiaTextRange::GetAttributeValue(int32_t textAttributeId) const
+    {
+        // Copied functionality from Types::UiaTextRange.cpp
+        if (textAttributeId == UIA_IsReadOnlyAttributeId)
+        {
+            return winrt::box_value(false);
+        }
+        else
+        {
+            return nullptr;
+        }
+    }
+
+    void XamlUiaTextRange::GetBoundingRectangles(com_array<double>& returnValue) const
+    {
+        returnValue = {};
+        try
+        {
+            SAFEARRAY* pReturnVal;
+            THROW_IF_FAILED(_uiaProvider->GetBoundingRectangles(&pReturnVal));
+
+            double* pVals;
+            THROW_IF_FAILED(SafeArrayAccessData(pReturnVal, (void**)&pVals));
+
+            long lBound, uBound;
+            THROW_IF_FAILED(SafeArrayGetLBound(pReturnVal, 1, &lBound));
+            THROW_IF_FAILED(SafeArrayGetUBound(pReturnVal, 1, &uBound));
+
+            long count = uBound - lBound + 1;
+
+            std::vector<double> vec;
+            vec.reserve(count);
+            for (int i = 0; i < count; i++)
+            {
+                double element = pVals[i];
+                vec.push_back(element);
+            }
+
+            winrt::com_array<double> result{ vec };
+            returnValue = std::move(result);
+        }
+        catch (...)
+        {
+        }

```

Comment:
you're not actually returning from this function!

Reply:
Isn't the out-param handled by line 76?

Reply:
Ugh yes I forgot this was the one stupid out-param function